src/main/java/Consolidator.java

```diff
@@ -15,43 +20,68 @@
     // for applying the mapping between the excel cells.
     private ArrayList<Integer> fromSheetIndices;
     private ArrayList<Integer> toSheetIndices;
-    private int indicesCouner = -1;
+    private ArrayList<Integer> toExcelIndices;
+    private int currentOutIdx = -1;
+    private int currentMappingIdx = -1;
 
     /**
      *
      * @param mappingFilePath a text file located at data/
      * @param inExcel
      * @param outExcels
      */
-    public Consolidator(String mappingFilePath, Excel inExcel, ArrayList<Excel> outExcels, int outExcelIdx) {
+    public Consolidator(String mappingFilePath, Excel inExcel, ArrayList<Excel> outExcels) {
         this.inExcel = inExcel;
         this.outExcels = outExcels;
         this.fromSheetIndices = new ArrayList<>();
         this.toSheetIndices = new ArrayList<>();
+        this.toExcelIndices = new ArrayList<>();
         this.cellMappings = new ArrayList<>();
 
         readFile(mappingFilePath);
-
-        // set sheet lookup index
-        // TODO: refactor this: Set index when creating the excel instances.
-        //  To do so, we have to change the control-flow of this class:
-        //  1. constructor should only load the mapping file and extract mapping information.
-        //  2. run the copy process by performing a method call.
-        updateSheetIndices(outExcelIdx);
-
-        ArrayList<CellMapping> sublistCellMappings = new ArrayList<>();
-        for (CellMapping mapping : cellMappings) {
-            if (mapping.getOutExcelFileIdx() == outExcelIdx) {
-                sublistCellMappings.add(mapping);
-            }
+        
+        int mappingIdx = 0;
+        for (int outExcelIdx: toExcelIndices) {
```

Comment:
Thanks a lot for implementing this - finally we do not have to hard-code the output files in the class `Gui`.  :smile:  :+1: